File: experiments/runs/2026-09-20_07-33-00_claim-office-java-example-mapping_exact-ptdd-v1-cc_opus-5-no-thinking/src/main/java/RiskSurcharges.java
/**
 * The MHPCO's risk surcharges. An item-specific modifier applies to the base
 * premium of the affected item, never to the policy total.
 */
final class RiskSurcharges {

    private static final double CURSE_SURCHARGE = 0.50;
    private static final double HIGH_ENCHANTMENT_SURCHARGE = 0.30;
    private static final int HIGH_ENCHANTMENT_LEVEL = 5;

    private RiskSurcharges() {
    }

    static double forItem(Item item) {
        double rate = 0;
        if (item.cursed()) {
            rate += CURSE_SURCHARGE;
        }
        if (isHighlyEnchanted(item)) {
            rate += HIGH_ENCHANTMENT_SURCHARGE;
        }
        return PriceList.basePremiumOf(item) * rate;
    }

    private static boolean isHighlyEnchanted(Item item) {
        return item.enchantment() >= HIGH_ENCHANTMENT_LEVEL;
    }
}

File: experiments/runs/2026-09-23_05-59-59_claim-office-example-mapping_exact-ptdd-v1-cc_opus-5-no-thinking/src/risk-surcharges.ts
import type { Item } from "./item.js";

const CURSE_SURCHARGE_RATE = 0.5;
const HIGH_ENCHANTMENT_SURCHARGE_RATE = 0.3;
const HIGH_ENCHANTMENT_THRESHOLD = 5;

function isCursed(item: Item): boolean {
  return item.cursed === true;
}

function isHighlyEnchanted(item: Item): boolean {
  return (item.enchantment ?? 0) >= HIGH_ENCHANTMENT_THRESHOLD;
}

export function riskSurchargeRate(item: Item): number {
  let rate = 0;
  if (isCursed(item)) {
    rate += CURSE_SURCHARGE_RATE;
  }
  if (isHighlyEnchanted(item)) {
    rate += HIGH_ENCHANTMENT_SURCHARGE_RATE;
  }
  return rate;
}
